SYCL/Reduction/reduction_span_pack.cpp

```diff
@@ -0,0 +1,194 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//
+// `Group algorithms are not supported on host device.` on Nvidia.
+// XFAIL: hip_nvidia
+
+// TODO: test disabled due to sporadic fails in level_zero:gpu RT.
+// UNSUPPORTED: linux && level_zero
+
+// This test performs basic checks of reductions initialized with a pack
+// containing at least one sycl::span
+
+#include <CL/sycl.hpp>
+using namespace sycl;
+
+int NumErrors = 0;
+
+template <int Dimensions> size_t getLinearSize(range<Dimensions> Range) {
+  return Range.size();
+}
+
+template <int Dimensions> size_t getLinearSize(nd_range<Dimensions> NDRange) {
+  return NDRange.get_global_range().size();
+}
+
+template <int Dimensions>
+size_t getLinearId(nd_range<Dimensions>, nd_item<Dimensions> Item) {
+  return Item.get_global_linear_id();
+}
+
+size_t getLinearId(range<1>, id<1> Id) { return Id[0]; }
+
+size_t getLinearId(range<2> Range, id<2> Id) {
+  return Id[0] * Range[1] + Id[1];
+}
+
+size_t getLinearId(range<3> Range, id<3> Id) {
+  return Id[0] * Range[1] * Range[2] + Id[1] * Range[2] + Id[2];
+}
+
+// Test a span and a regular sum
+template <size_t N, typename T, typename BinaryOperation, typename Range>
+void test1(queue Q, Range Rng, T Identity, T Value) {
+
+  // Initialize output to identity value
+  int *Sum = malloc_shared<int>(1, Q);
+  Q.single_task([=]() { *Sum = 0; }).wait();
+  T *Output = malloc_shared<T>(N, Q);
+  Q.parallel_for(range<1>{N}, [=](id<1> I) { Output[I] = Identity; }).wait();
+
+  // Perform generalized "histogram" with N bins
+  // TODO: Test Q.parallel_for when code_location is fixed
```

Comment:
The q.parallel_for issue is already resolved: https://github.com/intel/llvm/pull/4352

Reply:
Ah, thanks.  I've updated the test to run with both handler and queue in 1c63faba9cfcf94880d6e73c47001f966d0735ba.